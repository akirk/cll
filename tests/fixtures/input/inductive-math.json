[
	"Sure",
	"!",
	" Mathematical",
	" induction",
	" is",
	" a",
	" powerful",
	" technique",
	" used",
	" to",
	" prove",
	" statements",
	" about",
	" natural",
	" numbers",
	".\n\n",
	"Let's",
	" break",
	" it",
	" down",
	" into",
	" two",
	" main",
	" parts",
	":\n\n",
	"1",
	".",
	" **",
	"Base",
	" Case",
	"**",
	":",
	" First",
	",",
	" you",
	" show",
	" that",
	" the",
	" statement",
	" is",
	" true",
	" for",
	" the",
	" initial",
	" value",
	",",
	" typically",
	" \\(",
	" n",
	" =",
	" ",
	"1",
	" \\",
	").\n\n",
	"2",
	".",
	" **",
	"Ind",
	"uctive",
	" Step",
	"**",
	":",
	" Then",
	",",
	" you",
	" assume",
	" that",
	" the",
	" statement",
	" is",
	" true",
	" for",
	" some",
	" arbitrary",
	" natural",
	" number",
	" \\(",
	" k",
	" \\",
	")",
	" (",
	"this",
	" is",
	" your",
	" induct",
	"ive",
	" hypothesis",
	"),",
	" and",
	" you",
	" use",
	" that",
	" assumption",
	" to",
	" prove",
	" that",
	" the",
	" statement",
	" is",
	" also",
	" true",
	" for",
	" \\(",
	" k",
	" +",
	" ",
	"1",
	" \\",
	").\n\n",
	"If",
	" both",
	" of",
	" these",
	" steps",
	" are",
	" successfully",
	" proven",
	",",
	" you",
	" can",
	" conclude",
	" that",
	" the",
	" statement",
	" is",
	" true",
	" for",
	" all",
	" natural",
	" numbers",
	".\n\n",
	"To",
	" make",
	" this",
	" more",
	" concrete",
	",",
	" can",
	" you",
	" think",
	" of",
	" a",
	" specific",
	" statement",
	" or",
	" theorem",
	" you'd",
	" like",
	" to",
	" prove",
	" using",
	" induction",
	"?"
]
