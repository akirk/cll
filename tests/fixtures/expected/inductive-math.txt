Sure! Mathematical induction is a powerful technique used to prove statements about natural numbers.

Let's break it down into two main parts:

1. Base Case: First, you show that the statement is true for the initial value, typically n = 1.

2. Inductive Step: Then, you assume that the statement is true for some arbitrary natural number k (this is your inductive hypothesis), and you use that assumption to prove that the statement is also true for k + 1.

If both of these steps are successfully proven, you can conclude that the statement is true for all natural numbers.

To make this more concrete, can you think of a specific statement or theorem you'd like to prove using induction?